python/paddle/v2/fluid/backward.py

```diff
@@ -298,7 +298,8 @@ def _append_backward_ops_(block,
         # If the op has its own sub-block, deal with the sub-block first
         if op.has_attr("sub_block"):
             sub_block = program.block(op.block_attr("sub_block"))
-            grad_sub_block = program.create_block(parent_idx=sub_block.idx)
```

Comment:
Was this a bug?
`grad_sub_block = program.create_block(parent_idx=sub_block.idx)`
=>
`grad_sub_block = program.create_block(parent_idx=current_block.idx)`

Reply:
We use `parent` to specify the `forward block` of the `grad block` before.
Since we separate the `forward block` and `parent block`, here we change the parent of `grad block` to its actual parent.

Reply:
Excellent!

Reply:
Cool.